.pre-commit-config.yaml

```diff
@@ -33,3 +34,11 @@ repos:
     rev: v2.3.0
     hooks:
       - id: codespell
+  - repo: local
+    hooks:
+      - id: regenerate-files
+        name: regenerate generated files
+        language: system
+        entry: python src/trio/_tools/gen_exports.py
+        pass_filenames: false
+        files: ^src\/trio\/_core\/(_generated)?(_run|(_i(o_(common|epoll|kqueue|windows)|nstrumentation)))\.py$
```

Comment:
I feel like "any file with generated in it" is probably better and not overfitted to the current things trio supports.

Reply:
I could certainly change the regex to only match if a file has `_generated` in it, but I am trying to have it match both generated files and the files that are used to create the generated files, mostly because I ran into a case working on https://github.com/python-trio/trio/pull/3044 where the new ruff rules changed the source files and I didn't realize and didn't regenerate the generated files.

Reply:
Oh duh, that makes sense. I was thinking about this the wrong way around: you want to check for the inputs, not the outputs. Can we remove the `(_generated)?` bit entirely?

Reply:
Sure thing

Reply:
Changed with 78f93bdf